components/courses/app/models/courses/interview.rb

```diff
@@ -4,15 +4,16 @@ class Interview < ApplicationRecord
 
     self.table_name = 'courses_interviews'
 
-    belongs_to :season
     belongs_to :mentor
     belongs_to :student, optional: true
 
     validates :start_at, uniqueness: { scope: :mentor_id }, presence: true
-    validate  :half_hour_after_previous_interview
+    validate  :has_interview_earlier_than_allowed_interval
 
-    def half_hour_after_previous_interview
-      if mentor_has_interview_earlier_than_half_hour_before?
+    ALLOWED_INTERVAL = 30
+
+    def has_interview_earlier_than_allowed_interval
+      if Courses::Interview::HasInterviewEarlierThanAllowedInterval.new(mentor, self, ALLOWED_INTERVAL).call
         errors.add(:start_at, "should be at least 30 minutes after your previous interview")
```

Comment:
`"should be at least #{ALLOWED_INTERVAL} minutes after your previous interview"`